test/controllers/laboratories_controller_test.rb

```diff
@@ -0,0 +1,324 @@
+# frozen_string_literal: true
+
+require 'test_case'
+
+class LaboratoriesControllerTest < ActionController::TestCase
+  def setup; end
+
+  def teardown; end
+
+  test 'before action authenticate user' do
```

Comment:
I know this is basically ready to go, but could we add tests for the new before actions here and remove the redundant error checking ones below? I am doing the same right now for the vaccine delete

Reply:
I don't think it's redundant since it's making sure that those different before_actions are actually happening before each of the requested actions